File: Exercises2024/Chapter06_vle_free_energy/Fortran/1-Gibbs/Source/init_chem.f
      Subroutine Init_Chem(Switch)

C     ---Initialize And Calculate Chemical Potentials
C

      Implicit None

      Integer Switch, BoxID

      Include 'chem.inc'
      Include 'system.inc'

      If (Switch.Eq.0) Then
C        ---Initialize
         Do BoxID = 1, 2
            Chp(BoxID) = 0.0d0
            Ichp(BoxID) = 0
         End Do
      Else If (Switch.Eq.2) Then
C        ---Print Final Results
         Do BoxID = 1, 2
            If (Ichp(BoxID).Ne.0) Then
               Chp(BoxID) = -Dlog(Chp(BoxID)/Dble(Ichp(BoxID)))/Beta
            End If
         End Do
         Write (6, 99001) (Ichp(1)+Ichp(2))/2, Chp(1), Chp(2)
      Else
         Stop 'Error: Init_Chem'
      End If
99001 Format (' Chemical Potentials : ', /, ' Number Of Samples : ', 
     &        I12, /, ' Box 1 ', F7.3, /, ' Box 2 ', F7.3, /)
      Return
      End

File: Exercises2024/Chapter06_vle_free_energy/C/1-Gibbs/Source/init_chem.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include <math.h>
#include "system.h"
#include "chem.h"

int Init_Chem(int Switch)
{
  //  ---Initialize And Calculate Chemical Potentials
  int BoxID;
  
  if(Switch==0)
    {
      //    ---Initialize
      for(BoxID=0;BoxID<2;BoxID++)
	{   
	  Chp[BoxID] = 0.0;
	  Ichp[BoxID] = 0;
	}
    }
  else if(Switch==2)
    {
      //    ---Print Final Results
      for(BoxID=0;BoxID<2;BoxID++)
	{
	  if(Ichp[BoxID]!=0)
	    {
	      Chp[BoxID] = -log(Chp[BoxID]/(double)(Ichp[BoxID]))/Beta;
	    }
	}
      printf("------------------- \n");
      printf("Chemical Potentials \n");
      printf("Number of Samples: %d\n", ((Ichp[0]+Ichp[1])/2));
      printf("Box 1: %g\n",Chp[0]);
      printf("Box 2: %g\n",Chp[1]);
    }
   else
     {
       printf("Error: Init_Chem\n");
       exit(0);
     } 
  return 0;
}
